xilem_core/src/view.rs

```diff
@@ -271,3 +272,65 @@ where
         message_result
     }
 }
+
+impl<V: ?Sized> ViewMarker for Rc<V> {}
+/// An implementation of [`View`] which only runs rebuild if the states are different
+impl<State, Action, Context, Message, V> View<State, Action, Context, Message> for Rc<V>
+where
+    Context: ViewPathTracker,
+    V: View<State, Action, Context, Message> + ?Sized,
+{
+    type Element = V::Element;
+    type ViewState = ArcState<V::ViewState>;
+
+    fn build(&self, ctx: &mut Context) -> (Self::Element, Self::ViewState) {
+        let (element, view_state) = self.deref().build(ctx);
+        (
+            element,
+            ArcState {
```

Comment:
If this type can be used for both `Rc` and `Arc`, I'd argue the "atomic" part is irrelevant and shouldn't be part of its name, i.e. it should be renamed `RcState`.